src/Hamiltonians/abstract.jl

```diff
@@ -125,38 +131,53 @@ Stirling formula may be returned instead.
 ```jldoctest
 julia> dimension(HubbardMom1D(BoseFS((1,2,3))))
 28
-julia> dimension(HubbardMom1D(nearUniform(BoseFS{200,100})))
+julia> dimension(HubbardMom1D(near_uniform(BoseFS{200,100})))
+
 
-julia> dimension(Float64, HubbardMom1D(nearUniform(BoseFS{200,100})))
-1.3862737677578232e81
-julia> dimension(BigInt, HubbardMom1D(nearUniform(BoseFS{200,100})))
+julia> dimension(Float64, HubbardMom1D(near_uniform(BoseFS{200,100})))
+1.3862737677578234e81
+julia> dimension(BigInt, HubbardMom1D(near_uniform(BoseFS{200,100})))
 1386083821086188248261127842108801860093488668581216236221011219101585442774669540
 ```
 """
 function dimension(::Type{T}, ::BoseFS{N,M}) where {N,M,T<:Integer}
+    return try_binomial(T(N + M - 1), T(N))
+end
+function dimension(::Type{T}, ::BoseFS{N,M}) where {N,M,T<:AbstractFloat}
+    return approximate_binomial(T(N + M - 1), T(N))
+end
+function dimension(::Type{T}, f::FermiFS{N,M}) where {N,M,T<:Integer}
+    return try_binomial(T(M), T(N))
+end
+function dimension(::Type{T}, f::FermiFS{N,M}) where {N,M,T<:AbstractFloat}
+    return approximate_binomial(T(M), T(N))
+end
+function dimension(::Type{T}, b::BoseFS2C) where {T}
+    return dimension(T, b.bsa) * dimension(T, b.bsb)
+end
+function dimension(::Type{T}, c::CompositeFS) where {T}
+    return prod(x -> dimension(T, x), c.components)
+end
+
+function try_binomial(n::T, k::T) where {T}
     try
-        return T(binomial(T(N + M - 1), T(N)))
+        return T(binomial(n, k))
     catch
         return nothing
     end
 end
-function dimension(::Type{T}, ::BoseFS{N,M}) where {N,M,T<:AbstractFloat}
-    n = N + M - 1
-    k = N
+function approximate_binomial(n::T, k::T) where {T}
     try
         T(binomial(Int128(n), Int128(k)))
     catch
         T(exp(logbinomialapprox(n, k)))
     end
 end
-function dimension(::Type{T}, b::BoseFS2C) where T
-    return dimension(T, b.bsa) * dimension(T, b.bsb)
-end
```

Comment:
I think these methods should rather be defined in `BitStringAddresses`. I guess it would require moving the function there as well. However it can wait until the next PR where the function definition will go into the `Interfaces` module.